africanlii/forms.py

```diff
@@ -11,23 +12,29 @@ class BaseDocumentFilterForm(forms.Form):
     author = forms.CharField(required=False)
     doc_type = forms.CharField(required=False)
 
+    def __init__(self, data, *args, **kwargs):
+        self.params = QueryDict(mutable=True)
+        self.params.update(data)
+
+        super().__init__(self.params, *args, **kwargs)
+
     def filter_queryset(self, queryset, exclude=None):
 
-        year = self.cleaned_data.get("year")
+        year = self.params.getlist("year")
         alphabet = self.cleaned_data.get("alphabet")
-        author = self.cleaned_data.get("author")
-        doc_type = self.cleaned_data.get("doc_type")
+        author = self.params.getlist("author")
+        doc_type = self.params.getlist("doc_type")
 
         if year and exclude != "year":
-            queryset = queryset.filter(date__year=year)
+            queryset = queryset.filter(date__year__in=year)
 
         if alphabet and exclude != "alphabet":
             queryset = queryset.filter(title__istartswith=alphabet)
 
         if author and exclude != "author":
-            queryset = queryset.filter(author__name__iexact=author)
+            queryset = queryset.filter(author__name__in=author)
```

Comment:
@actlikewill can you pls double check that this is doing the right thing?

I also don't know if there's a way to let the form fields give us multiple choices for those fields (just curious -- this mechanism works too)